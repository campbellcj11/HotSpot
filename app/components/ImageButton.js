///////////////////////
/*

Author: ProjectNow Team
Class: ImageButton
Description: Custom Button.
Note:

*/
///////////////////////
import React, { Component } from 'react'
import {
  StyleSheet,
  Text,
  View,
  TouchableHighlight,
  Image,
} from 'react-native'

export default class ImageButton extends Component {
  state = {
    active: false,
  };

  _onHighlight = () => {
    this.setState({active: true});
  };

  _onUnhighlight = () => {
    this.setState({active: false});
  };

  render() {
    return (
      <TouchableHighlight
        onHideUnderlay={this._onUnhighlight}
        onPress={this.props.onPress}
        onShowUnderlay={this._onHighlight}
        style={[styles.button, this.props.style]}
        underlayColor="#FFFFFF00">
          <Image style={[styles.buttonImage,this.props.imageStyle]} source={this.props.image} resizeMode={'contain'}/>
      </TouchableHighlight>
    );
  }
}

var styles = StyleSheet.create({
  button: {
    justifyContent: 'center',
    alignItems: 'center',
  },
  buttonImage: {
  },
});
