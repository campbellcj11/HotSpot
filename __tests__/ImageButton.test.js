import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';

import ImageButton, { styles } from '../app/components/ImageButton';
import filterImage from '../app/images/filter.png'

const testWidth = 32;
const testHeight = 32;

it('renders a ImageButton using Snapshots', () => {
  expect(renderer.create(
      <ImageButton image={filterImage} style={{width:testWidth,height:testHeight}} imageStyle={{width:18,height:18,tintColor:'white'}}>
      </ImageButton>
  )).toMatchSnapshot();
});
