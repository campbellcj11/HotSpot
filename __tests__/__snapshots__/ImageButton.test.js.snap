exports[`test renders a ImageButton using Snapshots 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  hitSlop={undefined}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Array [
      Object {
        "backgroundColor": "transparent",
      },
      Array [
        Object {
          "alignItems": "center",
          "justifyContent": "center",
        },
        Object {
          "height": 32,
          "width": 32,
        },
      ],
    ]
  }
  testID={undefined}>
  <Image
    resizeMode="contain"
    source={1}
    style={
      Array [
        Object {},
        Object {
          "height": 18,
          "tintColor": "white",
          "width": 18,
        },
      ]
    } />
</View>
`;
